pkg/apis/logging/v1/output_types.go

```diff
@@ -136,3 +138,34 @@ type Elasticsearch struct {
 	// +optional
 	StructuredTypeName string `json:"structuredTypeName,omitempty"`
 }
+
+// Loki provides optional extra properties for `type: loki`
+type Loki struct {
+	// TenantKey is a meta-data key field to use as the TenantID,
+	// For example: 'TenantKey: kubernetes.namespace_name` will use the kubernetes
+	// namespace as the tenant ID.
+	//
+	// +optional
+	TenantKey string `json:"tenantKey,omitempty"`
```

Comment:
As mentioned on the enhancement doc, TenantKey should not be a key on one of the labels but rather an arbitrary string that forwarder users can pick based on their needs.